### PR TITLE
Fix load table might result table not exists

### LINKED ISSUE #393: 通过tispark查询某张表的时候提示table not exist
通过mysql-client查询某张表的时候是正常的
image
通过tispark查询的时候提示Table not exist
image
tispark的版本如下
image

### PR DESCRIPTION
**Pull request by birdstorm**
Fix #393 

And enclose a copy of conclusion:

>Bug in ScanIterator caused seldom table not exists.
>
>When currentCache is null means no keys found, whereas currentCache is empty means no values found. The difference lies in whether to continue scanning, because chances are that the same key is split in another region because of pending entries, region split, e.t.c.
>
>e.g., Assume we fetch table information in DB:123, ScanIterator sends a request containing an encoded key, say [1,2,3,0,0,0,1], and finds the first region that contains this key. If the region's range(e.g., [[1,2,3,0,0,0,0], [1,2,3,0,0,1,1])) was not enough to cover all keys [1,2,3,0,0,0,1], it is possible that it returns a partial result because some other table meta are contained in another region(e.g., [[1,2,3,0,0,1,1], [1,2,3,0,0,2])).
>
>Because ScanIterator was only used in fetching table meta info, tests not covering thousands of tables are unable to reveal this bug beforehand.

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java [hunk 1]
```diff
@@ -69,7 +69,11 @@ private boolean loadCache() {
     try (RegionStoreClient client = RegionStoreClient.create(region, store, session)) {
       BackOffer backOffer = ConcreteBackOffer.newScannerNextMaxBackOff();
       currentCache = client.scan(backOffer, startKey, version);
-      if (currentCache == null || currentCache.size() == 0) {
+      // currentCache is null means no keys found, whereas currentCache is empty means no values found
```
Comment (ilovesoup) [inline tikv-client/src/main/java/com/pingcap/tikv/operation/iterator/ScanIterator.java]: Please  add mock test case that covers empty region in b/w

File: core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala [hunk 2]
```diff
@@ -20,6 +20,25 @@ import org.apache.spark.sql.functions.{col, sum}
 
 class IssueTestSuite extends BaseTiSparkSuite {
 
+  val tableNum = 5000
```
Comment (ilovesoup) [inline core/src/test/scala/org/apache/spark/sql/IssueTestSuite.scala]: OK...Just leave this till we have more advanced mock framework.
